src/Fcf/Classes.hs

```diff
@@ -13,7 +13,24 @@ module Fcf.Classes
 
 import Fcf.Core
 
+-- $
+-- >>> import Fcf.Core
+-- >>> import Fcf.Utils
+-- >>> import Fcf.Data.Nat
+-- >>> import qualified GHC.TypeLits as TL
+
+
 -- | Type-level 'fmap' for type-level functors.
+--
+-- === __Example__
+--
+-- >>> data AddMul :: Nat -> Nat -> Exp Nat
+-- >>> type instance Eval (AddMul x y) = x TL.* x TL.+ 2 TL.* x TL.* y TL.+ y TL.* y
+-- >>> :kind! Eval (Map (AddMul 2) '[0, 1, 2, 3, 4])
+-- Eval (Map (AddMul 2) '[0, 1, 2, 3, 4]) :: [Nat]
+-- = '[4, 9, 16, 25, 36]
+--
+-- Note that the in example Map applies AddMul partially.
```

Comment:
I don't think this comment really adds much.

Reply:
ok removed